samza-core/src/main/java/org/apache/samza/operators/StreamGraphImpl.java

```diff
@@ -18,248 +18,120 @@
+  public Map<StreamSpec, InputStreamInternal> getInputStreams() {
+    return Collections.unmodifiableMap(inStreams);
   }
 
-  @Override public Map<StreamSpec, OutputStream> getOutStreams() {
-    Map<StreamSpec, OutputStream> outStreamMap = new HashMap<>();
-    this.outStreams.forEach((ss, entry) -> {
-        StreamSpec streamSpec = (entry instanceof IntermediateStreamImpl) ?
-          ((IntermediateStreamImpl) entry).getSpec() :
-          ((OutputStreamImpl) entry).getSpec();
-        outStreamMap.put(streamSpec, entry);
-      });
-    return Collections.unmodifiableMap(outStreamMap);
-  }
-
-  @Override
-  public StreamGraph withContextManager(ContextManager manager) {
-    this.contextManager = manager;
-    return this;
-  }
-
-  public int getNextOpId() {
-    return this.opId++;
+  public Map<StreamSpec, OutputStreamInternal> getOutputStreams() {
+    return Collections.unmodifiableMap(outStreams);
   }
 
   public ContextManager getContextManager() {
     return this.contextManager;
   }
 
-  /**
-   * Helper method to be get the input stream via {@link SystemStream}
-   *
-   * @param sstream  the {@link SystemStream}
-   * @return  a {@link MessageStreamImpl} object corresponding to the {@code systemStream}
-   */
-  public MessageStreamImpl getInputStream(SystemStream sstream) {
-    for (MessageStream entry: this.inStreams.values()) {
-      if (((InputStreamImpl) entry).getSpec().getSystemName().equals(sstream.getSystem()) &&
-          ((InputStreamImpl) entry).getSpec().getPhysicalName().equals(sstream.getStream())) {
-        return (MessageStreamImpl) entry;
-      }
-    }
-    return null;
-  }
-
-  <M> OutputStream<M> getOutputStream(MessageStreamImpl<M> intStream) {
-    if (this.outStreams.containsValue(intStream)) {
-      return (OutputStream<M>) intStream;
-    }
-    return null;
-  }
-
-  /**
-   * Method to generate intermediate stream from an operator ID.
-   *
-   * @param opId  operator ID
-   * @param parKeyFn  the function to extract the partition key from the input message
-   * @param <PK>  the type of partition key
-   * @param <M>  the type of input message
-   * @return  the {@link OutputStream} object for the re-partitioned stream
-   */
-  <PK, M> MessageStreamImpl<M> generateIntStreamFromOpId(int opId, Function<M, PK> parKeyFn) {
-    String opNameWithId = String.format("%s-%s", OperatorSpec.OpCode.PARTITION_BY.name().toLowerCase(), opId);
-    String streamId = String.format("%s-%s-%s",
-        config.get(JobConfig.JOB_NAME()),
-        config.get(JobConfig.JOB_ID(), "1"),
-        opNameWithId);
-    StreamSpec streamSpec = runner.getStreamSpec(streamId);
-
-    this.inStreams.putIfAbsent(streamSpec.getId(), new IntermediateStreamImpl(this, streamSpec, null, null, parKeyFn));
-    IntermediateStreamImpl intStream = (IntermediateStreamImpl) this.inStreams.get(streamSpec.getId());
-    this.outStreams.putIfAbsent(streamSpec.getId(), intStream);
-    return intStream;
+  int getNextOpId() {
```

Comment:
Feels a little boiler-platey in general, but makes sense to call it out explicitly for this method. Thanks.